lib/test.js

```diff
@@ -725,6 +725,7 @@ class Test {
 		this.clearTimeout();
 		this.verifyPlan();
 		this.verifyAssertions();
+		this.testPassed = !this.assertError;
 		await this.runTeardowns();
```

Comment:
I'd like to move the `error` and `passed` variable declarations up here, so we don't repeat the logic.

Reply:
The thing is that when you run teardowns, it might change the `this.assertionError` state.
So like
```javascript
		let error = this.assertError;
		let passed = !error;

		await this.runTeardowns();
// at this point if assertions made in teardowns and failed we'll not fail the test, so we have to do
                 error = this.assertError;
                 passed = !error;

		this.duration = nowAndTimers.now() - this.startedAt;
        
```